fulp_modules/QM_wrestling/code/wrestling.dm

```diff
@@ -0,0 +1,455 @@
+/*
+ *	The contents of this file were originally licensed under CC-BY-NC-SA 3.0 as part of Goonstation(https://ss13.co).
+ *	However, /tg/station and derivative codebases have been granted the right to use this code under the terms of the AGPL.
+ *	The original authors are: cogwerks, pistoleer, spyguy, angriestibm, marquesas, and stuntwaffle.
+ */
+
+/// Defines the Martial Art - Move to fulp_defines.dm once said file exists -Willard
+#define MARTIALART_FULPWRESTLING "Quartermaster Wrestling"
+/// How many steps it takes to throw the target - Used for throwing
+#define WRESTLING_THROW_STEPS 28
+
+
+/datum/martial_art/fulpwrestling
+	name = "Quartermaster Wrestling"
+	id = MARTIALART_FULPWRESTLING
+	help_verb = /mob/living/proc/FulpWrestling_help
+	smashes_tables = TRUE
+	var/datum/action/fulpslam/slam = new/datum/action/fulpslam()
+	var/datum/action/fulpdrop/drop = new/datum/action/fulpdrop()
+	var/list/valid_areas = list(/area/quartermaster)
+
+/datum/martial_art/fulpwrestling/proc/check_streak(mob/living/user, mob/living/target)
+	if(!can_use(user))
+		return FALSE
+	switch(streak)
+		if("fulpslam")
+			streak = ""
+			fulpslam(user,target)
+			return TRUE
```

Comment:
```suggestion
			if(user.pulling && user.pulling == target)
				streak = ""
				fulpslam(user,target)
				return TRUE
```
if you make this change, you should remove the pulling check in fulpslam()

Reply:
this'll make the slam much less janky to use- instead of needing to NOT have slam active, grab someone, activate slam mode, and then click on that person again, you can just have slam active and try to grab them twice